app/main.py

```diff
@@ -379,11 +379,12 @@ def url_exists(path, bucket_name=Config.DEFAULT_S3_BUCKET_NAME):
 
     query_args = request.args
     s3BucketName = query_args.get("s3BucketName", bucket_name)
+    s3_path = check_path_in_mangled_list(path)
```

Comment:
This does not need to go into the /s3-resource/ endpoint directly. The processing is done in the sort_files_by_mime_type method when we are getting the information for the list of object information

Reply:
It will also be good to squash the previous changes that have been discarded.

Reply:
@alan-wu Files are are also referenced from the sidebar. The file paths are contained in `scaffold-context-info.json` files

The interior of these context files is never seen by scicrunch and doesnt get processed in the sparc-api scicrunch processing

Reply:
> It will also be good to squash the previous changes that have been discarded.

Squashed them!

Reply:
> @alan-wu Files are are also referenced from the sidebar. The file paths are contained in `scaffold-context-info.json` files
> 
> The interior of these context files is never seen by scicrunch and doesnt get processed in the sparc-api scicrunch processing

Hmmm... thats true but I am also worry about leaving it in the s3-resource endpoint and we will never be able to test the wrongly mapped items (same paths but different datasets and one with name mangling and one without).

Reply:
@alan-wu what about if I add a flag that the sidebar calls to the route? 

ie:

`http://localhost:5000/s3-resource/328/1/files/derivative/mapped_Pig%207_thumbnail.jpeg?s3BucketName=prd-sparc-discover-use1&fixMagling=1`